### PR TITLE
BENCH-229-Fix-TestBase-abstract-methods

### PR DESCRIPTION
**Pull request by NogaAD**
Add TestCase to TestBase

### CODE REVIEW

File: answerking_app/tests/BaseTestClass.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
-from abc import abstractmethod
 from datetime import datetime, timedelta
+from unittest import TestCase
```
Comment (abbas-khan8) [inline answerking_app/tests/BaseTestClass.py]: Should this not be `from django.test import TestCase`?

**Review by JoeCSykes — APPROVED** [review]

File: answerking_app/tests/BaseTestClass.py (hunk 1 above)
Comment (NogaAD) [reply answerking_app/tests/BaseTestClass.py]: Changed to TransactionTestCase

File: answerking_app/tests/BaseTestClass.py [hunk 2]
```diff
@@ -1,6 +1,5 @@
-from abc import abstractmethod
 from datetime import datetime, timedelta
-
+from django.test import TransactionTestCase
```
Comment (PietroConvalleAD) [inline answerking_app/tests/BaseTestClass.py:2]: I'm fairly sure that TransactionTestCase should not be used if not dealing with a Database error

Comment (NogaAD) [reply answerking_app/tests/BaseTestClass.py:2]: Can you explain why that's an issue?
I think testCase had an issue with how we setup and teardown which TransactionTestCase doesn't.
"TransactionTestCase and TestCase are identical except for the manner in which the database is reset to a known state and the ability for test code to test the effects of commit and rollback"
from: https://docs.djangoproject.com/en/4.1/topics/testing/tools/#transactiontestcase

Comment (PietroConvalleAD) [reply answerking_app/tests/BaseTestClass.py:2]: mh, ok the only difference seems to be the speed

Comment (PietroConvalleAD) [reply answerking_app/tests/BaseTestClass.py:2]: I guess you can also remove the tear down and the 2 separate classes in each test file then

Comment (NogaAD) [reply answerking_app/tests/BaseTestClass.py:2]: Done

**Review by PietroConvalleAD — APPROVED** [review]

**Pull request merged**